lib/logstash/inputs/udp.rb

```diff
@@ -74,11 +72,13 @@ def udp_listener(output_queue)
       Thread.new { inputworker(i) }
     end
 
-    while true
+    while !stop?
       #collect datagram message and add to queue
       payload, client = @udp.recvfrom(@buffer_size)
```

Comment:
tcp, udp and a few others definitely need to be refactored to be non-blocking indeed, but I think it's out of scope for this, do you agree?